pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -2,24 +2,37 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
 import 'dart:io';
 
 import 'package:cli_config/cli_config.dart';
+import 'package:collection/collection.dart';
+import 'package:crypto/crypto.dart';
 import 'package:pub_semver/pub_semver.dart';
 
-import '../model/build_config.dart' as model;
-import '../model/build_mode.dart' as model;
-import '../model/ios_sdk.dart' as model;
-import '../model/link_mode_preference.dart' as model;
-import '../model/target.dart' as model;
+import '../model/metadata.dart';
+import '../model/target.dart';
+import '../utils/map.dart';
+import '../utils/yaml.dart';
+import 'architecture.dart';
+import 'asset.dart';
 import 'build_mode.dart';
 import 'ios_sdk.dart';
 import 'link_mode_preference.dart';
-import 'metadata.dart';
-import 'target.dart';
-
-abstract class BuildConfig {
-  /// The folder in which all output and intermediate artifacts should be
+import 'os.dart';
+
+part 'c_compiler_config.dart';
+part '../model/build_config.dart';
+part '../model/c_compiler_config.dart';
+
+/// The configuration for a `build.dart` invocation.
+///
+/// A package can choose to have a toplevel `build.dart` script. If such a
+/// script exists, it will be automatically run, by the Flutter and Dart SDK
+/// tools. The script will then be run with specific commandline arguments,
+/// which [BuildConfig] can parse and provide more convenient access to.
+abstract final class BuildConfig {
```

Comment:
Agreed with `Config`.

I had some thoughts about `Input` as well, but the C source files on disk are also input, so it's not the `BuildInput`.
If anything it would be the `BuildHookInput`, but that seems rather convoluted.

Agreed, this is not a context.

Let's stick with `BuildConfig` then.